sources/packages/backend/apps/workers/src/controllers/supporting-user/supporting-user.controller.ts

```diff
@@ -70,20 +81,27 @@ export class SupportingUserController {
       >
     >,
   ): Promise<MustReturnJobActionAcknowledgement> {
-    const supportingUser =
-      await this.supportingUserService.getSupportingUserById(
-        job.variables.supportingUserId,
-      );
-    if (!supportingUser) {
-      job.error(
-        SUPPORTING_USER_NOT_FOUND,
-        "Supporting user not found while checking for supporting user response.",
+    const jobLogger = new Logger(job.type);
+    try {
+      const supportingUser =
+        await this.supportingUserService.getSupportingUserById(
+          job.variables.supportingUserId,
+        );
+      if (!supportingUser) {
+        const message =
+          "Supporting user not found while checking for supporting user response.";
+        jobLogger.error(message);
+        job.error(SUPPORTING_USER_NOT_FOUND, message);
+      }
+      const outputVariables = filterObjectProperties(
+        supportingUser.supportingData,
+        job.customHeaders,
       );
+      jobLogger.log("Supporting user data loaded.");
+      return job.complete(outputVariables);
+    } catch (error: unknown) {
+      const errorMessage = `Unexpected error while loading supporting user data. ${error}`;
+      return createUnexpectedJobFail(errorMessage, job, jobLogger);
```

Comment:
I was expecting that we no longer need line 103.
The helper would be called as `return createUnexpectedJobFail(error, job, jobLogger);`